homeassistant/components/tile/device_tracker.py

```diff
@@ -107,23 +107,15 @@ def source_type(self):
         """Return the source type, eg gps or router, of the device."""
         return SOURCE_TYPE_GPS
 
-    @property
-    def state_attributes(self):
-        """Return the device state attributes."""
-        attr = {}
-        attr.update(
-            super().state_attributes,
-            **{
+    @callback
+    def _update_from_latest_data(self):
+        """Update the entity from the latest data."""
+        self._tile = self.coordinator.data[self._tile_uuid]
+        self._attrs.update(
```

Comment:
Looks like we can set these values in the `self._attrs` dict directly. We don't need to create a new dict and then update the existing `self._attrs` dict.

```py
self._attrs[ATTR_ALTITUDE] = self._tile["last_tile_state"]["altitude"]
...
```

Reply:
We use that pattern (`dict.update`) to add/update multiple `dict` keys in several places throughout the codebase; why not here?

Reply:
We don't need to create a new dictionary to change values of an existing dictionary. I can't speak for the existing code without explicit examples. There are probably many places where we could improve things.